### ISSUE TITLE
Temporal Dead Zone is not catched

### ISSUE THREAD

**Issue by vitaly-zdanevich** [eslint/eslint#10375]
* **ESLint Version:** 4.19.1
* **Node Version:** 9.8.0
* **npm Version:** 6.0.1

**What parser (default, Babel-ESLint, etc.) are you using?**
Default.

**Please show your full configuration:**
```
module.exports = {
    "env": {
        "browser": true,
        "es6": true
    },
    "parserOptions": {
        "ecmaVersion": 2017,
        "ecmaFeatures": {
            "experimentalObjectRestSpread": true
        }
    },
    "extends": "eslint:recommended",
    "globals": {
        "chrome": false
    },
    "rules": {
        "indent": [
            "error", 4
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ],
        "no-console": "off",
        "no-unused-vars": [1,
            {"argsIgnorePattern": "_"}]
    }
};
```

Code is:
```js
const a = b * 2;

const b = 1;

console.log(a);
```
No errors from eslint, but execution will produce `ReferenceError`.

**Comment by vitaly-zdanevich** [comment]
I found that I need to use [no-use-before-define](https://eslint.org/docs/rules/no-use-before-define) - but I think that this behavior must be default for let/const.

**Comment by not-an-aardvark** [comment]
Hi, sorry about the delayed response here.

As you found out, the correct way to report this issue is to use the `no-use-before-define` rule. ESLint doesn't have any rules enabled by default, so I'm not sure it would make sense to report an error by default for this case.
